### PR TITLE
Mw fixes ksp120

### PR DESCRIPTION
**Pull request by mwerle**
Various fixes to get KIS compiled under KSP 1.2; no detailed testing as yet.

### CODE REVIEW

**Comment by ihsoft** [comment]
Is any of these changes **required** to run under 1.2? Or they are just cleanups?

UPD. Nevermind. Figured out you mentioned it in the request description. Thanks! I'll use the changes once ready to go with 1.2.

**Comment by mwerle** [comment]
The only one I'm not sure about is the fix for the science data transmission value. The original API used no longer exists but not sure if my change calculates the correct value.

File: Plugins/Source/KISAddonCursor.cs
Comment (ihsoft) [inline Plugins/Source/KISAddonCursor.cs:86]: This is awkward. It's a static member, it should get initialized before anything else is triggered. If it doesn't (e.g. due to KSP cloning) then member must be turned into instance member.

File: Plugins/Source/KISAddonPickup.cs
Comment (ihsoft) [inline Plugins/Source/KISAddonPickup.cs:880]: Please, in the following changes stick with the code style (2 spaces indentation, no wrapping for opening bracket, etc.). I will change these ones myself once merge is done but it's better to not do second run on the code.

File: Plugins/Source/KISAddonPointer.cs
Comment (ihsoft) [inline Plugins/Source/KISAddonPointer.cs]: No need to keep commented code.

File: Plugins/Source/KISAddonCursor.cs
Comment (mwerle) [reply Plugins/Source/KISAddonCursor.cs:86]: Awake () is run as soon as the script is loaded (according to Unity docs) and one of the APIs used on that line threw an error as not being allowed to run in ctors/initializers.

Can change the member to instance but then if there's any chance of the instance getting destructed might also need to manually Dispose of the resource. 

Conversely since there's only once we load the script and set the static even though we may have multiple instances they'll all share the member keeping the pre-1.2 logic. I don't know KIS well enough to know if this is a singleton or not.

File: Plugins/Source/KISAddonPickup.cs
Comment (mwerle) [reply Plugins/Source/KISAddonPickup.cs:880]: Sorry, thought I had done.  Been fighting with monodevelop to stop screwing around with my code formatting.  I'll be more vigilant in future before submitting.

File: Plugins/Source/KISAddonPointer.cs
Comment (mwerle) [reply Plugins/Source/KISAddonPointer.cs]: I left it as there was a double-init for audioBipWrong : the initializer was "new AudioSource()" (which raises an error), and Awake() already had the current code for this member. So I'm not sure why that field was being initialized twice; figured the maintainer could then make an executive decision on this.

File: Plugins/Source/KISAddonCursor.cs
Comment (ihsoft) [reply Plugins/Source/KISAddonCursor.cs:86]: Disposing is a problem anyways. Every instantiation of the cursor will create new instance of `GUIStyle`, and old one will be just abandoned. Though, I don't think this kind of resource needs disposing.

That said, converting into instance member should be OK.

File: Plugins/Source/KISAddonPointer.cs
Comment (ihsoft) [reply Plugins/Source/KISAddonPointer.cs]: Comments get lost in the code very quickly since it's unusual for people to review own code once it's submitted :) If mistake is not caught during the review then it turns into bug.

**Review by ihsoft — APPROVED** [review]

**Pull request merged**
